src/NodeVisitor/YieldExpressionVisitor.php

```diff
@@ -25,7 +25,7 @@ public function enterNode(Node $node)
 
         $valueClass = get_class($node->value);
         if (isset($this->lowerPrecedenceExpressionClasses[$valueClass])) {
-            $this->addContextMessage(
+            $this->addContextError(
                 'Yielding expression with precedence lower than "yield"',
```

Comment:
If indirect variable access is considered a warning, then this and the `YieldInExpressionContextVisitor` should probably emit warnings too. This yield usage is completely legal, it just has different meaning in PHP 7